device/tt_silicon_driver.cpp

```diff
@@ -3025,11 +3098,40 @@ int tt_SiliconDevice::iatu_configure_peer_region (int logical_device_id, uint32_
     struct PCIdevice* pci_device = get_pci_device(logical_device_id);
     auto architecture_implementation = pci_device->hdev->get_architecture_implementation();
 
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 0 * 4, region_id_to_use);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 1 * 4, dest_bar_lo);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 2 * 4, dest_bar_hi);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 3 * 4, region_size);
-    arc_msg(logical_device_id, 0xaa00 | architecture_implementation->get_arc_message_setup_iatu_for_peer_to_peer(), true, 0, 0);
+    // BR: ARC doesn't work yet on Blackhole, so programming ATU directly. Should be removed when arc starts working.
```

Comment:
Maybe add TODO to this comment. It will be more clear for others looking into this that this needs to be fixed when ARC is enabled